### PR TITLE
[benchmark] Add super resolution

### PR DESCRIPTION
**Pull request by cuiyanx**
Signed-off-by: cuiyan <yanx.cui@intel.com>

### CODE REVIEW

**Comment by cuiyanx** [comment]
@BruceDai please review

**Comment by cuiyanx** [comment]
Fix: #828

File: benchmark/utils_sr.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+  /**
+   * Setup model
+   * @returns {Promise<void>}
+   */
+  async setupAsync() {
+    await this.setInputOutput();
+    let backend = this.backend.replace('WebNN', 'WebML');
+    const modelName = this.modelInfoDict.modelName;
+    let loadResult = await loadModelAndLabels(this.modelInfoDict.modelFile, null);
```
Comment (BruceDai) [inline benchmark/utils_sr.js]: @cuiyanx  I suggest to update function loadModelAndLabels using default parameter label=null, what do you think？

Comment (cuiyanx) [reply benchmark/utils_sr.js]: @BruceDai Agree!

File: benchmark/utils_sr.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  async executeSingleAsync() {
+    let result = await this.model.compute([this.inputTensor],
+                                          [this.outputTensor]);
+    console.log(`compute result: ${result}`);
+  }
+
+  async executeAsync() {
+    let results = [];
+    for (let i = 0; i < this.iterations; i++) {
+      this.onExecuteSingle(i);
```
Comment (BruceDai) [inline benchmark/utils_sr.js]: Please add this code 
```javascript
await new Promise(resolve => requestAnimationFrame(resolve));
```
after 
```javascript
this.onExecuteSingle(i);
```
to update execution process status likes **Iteration: <i> / <iterations>** on UI

There is the same issue in benchmark/utils_ic.js, please help fix it toghter, thanks

**Comment by cuiyanx** [comment]
@BruceDai please review

File: benchmark/utils_sr.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
+// Benchmark for Super Resolution models
+class SRBenchmark extends Benchmark {
+  constructor(modelName, backend, iterations) {
+    super(...arguments);
+    this.modelInfoDict = getModelInfoDict(superResolutionModels, modelName);
+    this.model = null;
+    this.inputTensor = null;
+    this.outputTensor = null;
+    this.outputElement = document.getElementById('superCanvas');
+    this.inCanvas = document.createElement('canvas');
```
Comment (BruceDai) [inline benchmark/utils_sr.js]: @cuiyanx  Those 4 members inCanvas/inCtx  / outCanvas/outCtx aren't needed, they can be declared when using.
 Please move inCanvas/inCtx into function setInputOutput(), and move outCanvas/outCtx into function drawOutput(canvas).

**Comment by BruceDai** [comment]
LGTM, thanks @cuiyanx 
@ibelem Please help merge this PR, thanks

**Comment by BruceDai** [comment]
@cuiyanx There is a little issue with picking image, please fix it.

**Comment by cuiyanx** [comment]
@BruceDai please review

**Comment by BruceDai** [comment]
LGTM, thanks @cuiyanx 
@ibelem Please help merge this PR, thanks

File: benchmark/ui.js [hunk 4]
```diff
@@ -35,13 +36,44 @@ function updateOpsSelect() {
...
+        srCanvas.height = 96;
+        break;
+      case 'SRGAN 128x4 (TFLite)':
+        srCanvas.width = 128;
+        srCanvas.height = 128;
+        break;
+      default:
+        srCanvas.width = 96;
+        srCanvas.height = 96;
+    }
```
Comment (ibelem) [inline benchmark/ui.js]: Please, i prefer to use CSS rather than javascript for this kind of work, thoughts?

File: benchmark/ui.js [hunk 5]
```diff
@@ -35,13 +36,44 @@ function updateOpsSelect() {
...
+    let imageBytes = await loadImage(imageElement.src);
+    srCtx.drawImage(imageBytes, 0, 0, srCanvas.width, srCanvas.height);
+    imageElement.src = srCanvas.toDataURL();
+  } else {
+    imageElement.parentNode.setAttribute('align', 'center');
```
Comment (ibelem) [inline benchmark/ui.js]: Also, better to have the align: center in css.

**Comment by cuiyanx** [comment]
@ibelem I am rebased this pr and fix the conflict, please review

**Comment by ibelem** [comment]
Thanks @cuiyanx . As i suggested, more improves including file/patch structures required in new PRs.

**Pull request merged**